autopilot/hostscore.go

```diff
@@ -0,0 +1,256 @@
+package autopilot
```

Comment:
The score code could be isolated into a separate PR. This PR is already quite long and will only get longer with testing.

Reply:
jup, good idea